bench/bench.ml

```diff
@@ -95,17 +95,20 @@ let dune_build () =
   let stdout_to = Process.Io.make_stdout Swallow in
   let stderr_to = Process.Io.make_stderr Swallow in
   let open Fiber.O in
+  (* Build with timings *)
   let+ times =
     Process.run_with_times dune ~display:Quiet ~stdin_from ~stdout_to ~stderr_to
       [ "build"; "@install"; "--release" ]
   in
-  times.elapsed_time
+  (* Run GC and compactify before querying GC stats *)
+  Gc.full_major ();
```

Comment:
You're handling the GC in the wrong process here. It's the dune command above that should spit out its GC stats.

To do this, you could add some `--dump-gc-stats-end=FILE` argument that would add an `at_exit` hook to dump this info.

Reply:
I've added that option in #8072 (which is currently also included here). The data I'm getting now makes much more sense. I'll try to test out your PRs again.